Mathlib/NumberTheory/GaussSum.lean

```diff
@@ -227,7 +226,7 @@ theorem Char.card_pow_card {F : Type*} [Field F] [Fintype F] {F' : Type*} [Field
   have := Fact.mk (hchar.subst hp')
   rw [Ne, ← Nat.prime_dvd_prime_iff_eq hp' hp, ← isUnit_iff_not_dvd_char, hchar] at hch₁
   exact Char.card_pow_char_pow (hχ₂.comp _) ψ.char (ringChar FF') n' hch₁ (hchar ▸ hch₂)
-       (gaussSum_sq (hχ₁.comp <| RingHom.injective _) (hχ₂.comp _) ψ.prim)
+       (gaussSum_sq ((ringHomComp_ne_one_iff (RingHom.injective _)).mpr hχ₁) (hχ₂.comp _) ψ.prim)
```

Comment:
Dot notation was a benefit of `IsNontrivial` ...

Reply:
Indeed, that's true; but there's a tradeoff against having to cook up an `IsNontrivial` type when you want to apply the lemmas. In my ongoing Dirichlet-L-series project I found `\chi \ne 1` to be an easier hypothesis to work with, and Yael seems to have found the same in the add-char context.